cachi2/core/package_managers/bundler/parser.py

```diff
@@ -0,0 +1,160 @@
+
+    Attributes:
+        source:     The source URL of the gem.
+        checksum:   The checksum of the gem.
+    """
+
+    source: str
+    checksum: Optional[str] = None
+
+    @property
+    def purl(self) -> str:
+        """Get purl for this gem dependency."""
+        return NotImplemented
+
+
+class GitDependency(_GemMetadata):
+    """
+    Represents a git dependency.
+
+    Attributes:
+        url:        The URL of the git repository.
+        revision:   Commit hash.
+        branch:     The branch of the git repository.
+    """
+
+    url: str
+    revision: str
+    branch: Optional[str] = None
+
+    @property
+    def purl(self) -> str:
+        """Get purl for this git dependency."""
+        return NotImplemented
+
+    @pydantic.field_validator("url")
+    def validate_git_url(cls, url: str) -> str:
+        """Validate that the git URL is a valid git URL."""
+        if not url.startswith("https://"):
+            raise ValueError("GIT dependencies must use HTTPS protocol")
+
+        return url
+
+    @pydantic.field_validator("revision")
+    def validate_git_revision(cls, revision: str) -> str:
+        """Validate that the git revision is a valid git ref."""
+        git_ref_format = re.compile(r"^[a-fA-F0-9]{40}$")
+        if not git_ref_format.match(revision):
+            raise ValueError("GIT dependencies must have a valid revision")
+
+        return revision
+
+
+class PathDependency(_GemMetadata):
+    """
+    Represents a path dependency.
+
+    Attributes:
+        path:       Subpath from package root.
+    """
+
+    path: str
+
+    @property
+    def purl(self) -> str:
+        """Get purl for this path dependency."""
+        return NotImplemented
+
+
+BundlerDependency = Union[GemDependency, GitDependency, PathDependency]
+ParseResult = list[BundlerDependency]
+
+
+def parse_gemlock(package_dir: RootedPath) -> ParseResult:
+    """Parse a Gemfile.lock file and return a list of dependencies."""
+    gemfile_lock_path = package_dir.join_within_root(GEMFILE_LOCK)
+    if not gemfile_lock_path.path.exists():
+        reason = f"Gemfile.lock not found at {gemfile_lock_path.path}"
+        solution = "Please ensure the Gemfile.lock file is present in the package directory"
+        raise PackageRejected(reason=reason, solution=solution)
+
+    # TODO: Merge both conditions ?
+
+    gemfile_path = package_dir.join_within_root(GEMFILE)
+    if not gemfile_path.path.exists():
+        reason = f"Gemfile not found at {gemfile_path.path}"
+        solution = "Please ensure the Gemfile is present in the package directory"
+        raise PackageRejected(reason=reason, solution=solution)
+
+    bundler_package = files("cachi2.core.package_managers.bundler")
+    lockfile_parser_script = bundler_package.joinpath("lockfile-parser.rb")
+
+    output = subprocess.run(
+        args=[str(lockfile_parser_script)],
+        cwd=package_dir.path,
+        capture_output=True,
+        shell=True,
```

Comment:
Why do you need a shell if you're only passing path to the script itself in args and the script is marked executable? We should avoid the need for a shell whenever possible.

Reply:
Bandit is complaining about it.  We should probably disable this rule globally since I saw it [here](https://github.com/containerbuildsystem/cachi2/blob/b737d90806a72977d1aa664ea470d963c5248375/cachi2/core/utils.py#L51) as well.

Reply:
Yes, we need to disable it, because the rule is completely dumb - not only will it not accept a string literal passed directly (what more can you do?), but it also suggests usage of a known attack vector. To support my argument against usage of `shell=true`, have a look at this example: https://security.openstack.org/guidelines/dg_use-subprocess-securely.html

Last but not least this also goes along another of my recent comments in a different PR where I said we ought not blindly follow all rules coming from the linters and instead use common sense; this is a prime example of where things can go utterly wrong.

Reply:
Aand even Ruff documents it as bad (if I understood correctly the whole page): https://docs.astral.sh/ruff/rules/subprocess-without-shell-equals-true/